src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchCheck.java

```diff
@@ -212,16 +212,9 @@ public void visitToken(DetailAST detailAST) {
     private static List<DetailAST> getAllExceptionTypes(DetailAST parentToken) {
         DetailAST currentNode = parentToken.getFirstChild();
         final List<DetailAST> exceptionTypes = new LinkedList<>();
-        if (currentNode.getType() == TokenTypes.BOR) {
-            exceptionTypes.addAll(getAllExceptionTypes(currentNode));
-            currentNode = currentNode.getNextSibling();
+        while (currentNode != null) {
```

Comment:
Please remove `BOR` from the Javadoc of this method.

Reply:
Or, even better, remove this method. It was a workaround for the bug you are fixing and does nothing for now.
The method `visitToken` should use `getFirstChild`/`getNextSibling` directly.

Reply:
Added some more `illegalClassNames` to show that we use this iterative approach to getting Exception class names: https://github.com/checkstyle/checkstyle/pull/10136/files#r659173444

Reply:
I'm ok to keep this method, I just want to note that it copies from one linked list to another.

Reply:
I totally misunderstood what you meant; I have removed this method.  Thanks for pointing this out :)